### PR TITLE
Link checker: Add a method to ignore non-API docs for objects.inv

### LINKED ISSUE #2296: Improve link checker ignores of objects.inv for non-API docs
We ignore all of these because the docs have non-API docs in their Sphinx build, like tutorials, so the objects.inv has a ton of invalid links.

https://github.com/Qiskit/documentation/blob/6999dcabd9eb3144238aac8ed6fc731ec134bc65/scripts/js/lib/links/ignores.ts#L17-L27

However, it would be better to still check objects.inv and not ignore the API links. We should automatically do that rather than making you specify the non-API links in objects.inv.

### PR DESCRIPTION
**Pull request by arnaucasau**
This PR adds a new mechanism to ignore links where we can add a regex expression to ignore all the URLs that match it. It adds regexes for the non-API docs in the addons' and runtime's objects.inv.

Closes #2296

### CODE REVIEW

File: scripts/js/lib/links/ignores.ts [hunk 1]
```diff
@@ -79,6 +68,52 @@ export const ALWAYS_IGNORED_URLS = new Set([
...
+  return ["how-tos", "how_tos", "install", "index", "explanations"]
+    .map((path) => [
+      // Latest version
+      `\/api\/qiskit-addon-[^\/]+\/${path}(\/.*|#.*|$)`,
+      // Historical versions
+      `\/api\/qiskit-addon-[^\/]+\/[0-9]+\.[0-9]\/${path}(\/.*|#.*|$)`,
+      // Dev version
+      `\/api\/qiskit-addon-[^\/]+\/dev\/${path}(\/.*|#.*|$)`,
+    ])
+    .flat();
```
Comment (Eric-Arellano) [inline scripts/js/lib/links/ignores.ts:87]: You can use `flatMap()` instead to avoid this part :)

Comment (Eric-Arellano) [reply scripts/js/lib/links/ignores.ts:87]: Also addons don't have dev docs

File: scripts/js/lib/links/ignores.ts [hunk 2]
```diff
@@ -79,6 +68,52 @@ export const ALWAYS_IGNORED_URLS = new Set([
...
+    "primitives",
+    "compare",
+    "retired",
+  ]
+    .map((path) => [
+      `\/api\/qiskit-ibm-runtime\/0.16\/${path}(\/.*|#.*|$)`,
+      `\/api\/qiskit-ibm-runtime\/0.15\/${path}(\/.*|#.*|$)`,
+      `\/api\/qiskit-ibm-runtime\/0.14\/${path}(\/.*|#.*|$)`,
+    ])
+    .flat();
```
Comment (Eric-Arellano) [inline scripts/js/lib/links/ignores.ts:109]: Ditto flatmap

File: scripts/js/lib/links/ignores.ts [hunk 3]
```diff
@@ -79,6 +68,52 @@ export const ALWAYS_IGNORED_URLS = new Set([
...
+    "index",
+    "sessions",
+    "primitives",
+    "compare",
+    "retired",
+  ]
+    .map((path) => [
+      `\/api\/qiskit-ibm-runtime\/0.16\/${path}(\/.*|#.*|$)`,
+      `\/api\/qiskit-ibm-runtime\/0.15\/${path}(\/.*|#.*|$)`,
+      `\/api\/qiskit-ibm-runtime\/0.14\/${path}(\/.*|#.*|$)`,
```
Comment (Eric-Arellano) [inline scripts/js/lib/links/ignores.ts:107]: I think you could use `{14,15,16}` to consolidate these into one rule?

**Review by Eric-Arellano — COMMENTED** [review]
Nice work!

File: scripts/js/lib/links/ignores.ts (hunk 2 above)
Comment (arnaucasau) [reply scripts/js/lib/links/ignores.ts]: I removed the `flat` call in this case given that it's not necessary anymore after combining the three regexes into one

**Review by Eric-Arellano — APPROVED** [review]

**Pull request merged**